snuba/web/split.py

```diff
@@ -328,6 +328,12 @@ def execute(
             metrics.increment("column_splitter.query_above_limit")
             return None
 
+        # Do not split if there is already a condition on an ID column
+        if self.__id_column in [
+            col.column_name for col in query.get_columns_referenced_in_conditions_ast()
+        ]:
```

Comment:
I remember eventstore using conditions like 
```
[
        [TIMESTAMP, ">=", event.timestamp],
        [[TIMESTAMP, ">", event.timestamp], [EVENT_ID, ">", event.event_id]],
    ]
```

That condition on event_id would block the splitter here. Is that intentional ?

Reply:
Oh damn, I forgot about this scenario and it's not intentional. What do you think about only skipping the splitter if the operator is `=` or `IN`?

Reply:
seems good to me.
I would advise you to use the matcher api to find suitable conditions.
https://github.com/getsentry/snuba/blob/master/snuba/query/matchers.py#L81

Define a pattern that matches a FunctionCall with function_name "equals" or "in". and first parameter a column with name "event_id".
Then iterate over all the expressions in the query and match it against every expression.

Reply:
Ok, hopefully I used the matcher API correctly 😀. This should also check the operator now.